cylc/flow/cycling/integer.py

```diff
@@ -27,7 +27,7 @@
 )
 
 CYCLER_TYPE_INTEGER = "integer"
-CYCLER_TYPE_SORT_KEY_INTEGER = "a"
+CYCLER_TYPE_SORT_KEY_INTEGER = 0
```

Comment:
Sorting by integer seems more natural than by string

Reply:
We could conceivably allow other cycling types in the future (e.g. to cycle over a list of named datasets). They could all  be represented as strings, so long as integer cycles are also represented as strings (but compared as integers). The Point and Interval base class docstrings in `cylc/flow/cycling/__init__.py` say that derived classes should be "string based".  

However, (as your comment refers to?) this TYPE_SORT_KEY seems to be used as a proxy for instance comparison when comparing points of different cycling types ... and I don't understand why we'd want to do that either.  Any thoughts @oliver-sanders ? (`git blame` points to Ben on this, but I think it predates your involvement...).

Reply:
Yes, I changed both integer's & iso8601's `TYPE_SORT_KEY`s to ints. They are used here
https://github.com/cylc/cylc-flow/blob/87e22ab623ccccb48f990dd9d64357453c1c8a56/cylc/flow/cycling/__init__.py#L110-L118